### PR TITLE
Added ITaskbarList and ITaskbarList2 in shobjidl

### PR DESCRIPTION
**Pull request by edwin0cheng**
This PR added 2 Com Interface ITaskbarList and ITaskbarList2 in shobjidl.

### CODE REVIEW

File: src/um/shobjidl.rs [hunk 1]
```diff
@@ -17,6 +17,9 @@ use um::shtypes::COMDLG_FILTERSPEC;
 use um::unknwnbase::{IUnknown, IUnknownVtbl};
 use um::winnt::{HRESULT, LPCWSTR, LPWSTR, WCHAR};
 
+DEFINE_GUID!{CLSID_TaskbarList,
+    0x56fdf344, 0xfd6d, 0x11d0, 0x95, 0x8a, 0x00, 0x60, 0x97, 0xc9, 0xa0, 0x90}
```
Comment (Connicpu) [inline src/um/shobjidl.rs]: CLSID_TaskbarList should be defined in shobjidl_core.rs

File: src/um/shobjidl.rs [hunk 2]
```diff
@@ -382,3 +385,31 @@ interface IFileDialog2(IFileDialog2Vtbl): IFileDialog(IFileDialogVtbl) {
         psi: IShellItem,
     ) -> HRESULT,
 });
+
```
Comment (Connicpu) [inline src/um/shobjidl.rs]: No extra newlines

File: src/um/shobjidl.rs [hunk 3]
```diff
@@ -382,3 +385,31 @@ interface IFileDialog2(IFileDialog2Vtbl): IFileDialog(IFileDialogVtbl) {
...
+
+RIDL!(
+#[uuid(0x602d4995, 0xb13a, 0x429b, 0xa6, 0x6e, 0x19, 0x35, 0xe4, 0x4f, 0x43, 0x17)]
+interface ITaskbarList2(ITaskbarList2Vtbl): ITaskbarList(ITaskbarListVtbl) {
+    fn MarkFullscreenWindow(
+        hwnd: HWND,
+        fFullscreen: BOOL,
+    ) -> HRESULT,    
+});
```
Comment (Connicpu) [inline src/um/shobjidl.rs]: Don't forget a newline at the end of the file

**Review by Connicpu — CHANGES_REQUESTED** [review]
All of the definitions you added actually belong in shobjidl_core.rs, no shobjidl.rs

File: src/um/shobjidl.rs [hunk 4]
```diff
@@ -382,3 +385,31 @@ interface IFileDialog2(IFileDialog2Vtbl): IFileDialog(IFileDialogVtbl) {
         psi: IShellItem,
     ) -> HRESULT,
 });
+
+RIDL!(
```
Comment (retep998) [inline src/um/shobjidl.rs]: Item level macros should use `{}` and there should be no newline between `RIDL!{` and `#[uuid]`.

File: src/um/shobjidl.rs (hunk 1 above)
Comment (edwin0cheng) [reply src/um/shobjidl.rs]: @Connicpu so the GUID , ITaskbarList and ITaskbar2List both should added belong to shobjidl_core.rs, not only the GUID, right ?

Comment (retep998) [reply src/um/shobjidl.rs]: Yes.

**Comment by edwin0cheng** [comment]
It should be fine now :) @Connicpu

**Comment by Connicpu** [comment]
Looks like it isn't passing CI. Make sure you run `cargo test --features shobjidl_core` before pushing

**Comment by edwin0cheng** [comment]
Sorry my bad, just fixed and tested this time.

**Comment by Connicpu** [comment]
@edwin0cheng Ah sorry, last little thing! You never fixed the whitespace issues from retep998's comment on your PR

File: src/um/shobjidl_core.rs [hunk 5]
```diff
@@ -71,3 +73,29 @@ interface IModalWindow(IModalWindowVtbl): IUnknown(IUnknownVtbl) {
 //22307
 //27457
 pub type IShellItemFilter = IUnknown; // TODO
+
+RIDL!{
```
Comment (retep998) [inline src/um/shobjidl_core.rs]: `#[uuid]` should go on the same line as `RIDL!{`

**Review by retep998 — CHANGES_REQUESTED** [review]

**Comment by edwin0cheng** [comment]
Fixed :)

File: src/um/shobjidl_core.rs [hunk 6]
```diff
@@ -7,12 +7,14 @@
...
-use shared::minwindef::ULONG;
-use shared::windef::HWND;
+use shared::minwindef::{BOOL, ULONG};
+use shared::windef::{HWND};
 use um::objidl::IBindCtx;
 use um::unknwnbase::{IUnknown, IUnknownVtbl};
 use um::winnt::{HRESULT, LPWSTR};
 
+DEFINE_GUID!{CLSID_TaskbarList,
+0x56fdf344, 0xfd6d, 0x11d0, 0x95, 0x8a, 0x00, 0x60, 0x97, 0xc9, 0xa0, 0x90}
```
Comment (retep998) [inline src/um/shobjidl_core.rs]: The second line of a `DEFINE_GUID` should be indented.

**Review by retep998 — CHANGES_REQUESTED** [review]

**Comment by edwin0cheng** [comment]
Indent fixed 👍

**Review by retep998 — APPROVED** [review]

**Pull request merged**